runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -63,13 +63,13 @@ GC_ArrayletObjectModel::AssertArrayletIsDiscontiguous(J9IndexableObject *objPtr)
 void
 GC_ArrayletObjectModel::AssertContiguousArrayletLayout(J9IndexableObject *objPtr)
 {
-        Assert_MM_true(InlineContiguous == getArrayLayout(objPtr));
+	Assert_MM_true(InlineContiguous == getArrayLayout(objPtr));
 }
 
 void
 GC_ArrayletObjectModel::AssertDiscontiguousArrayletLayout(J9IndexableObject *objPtr)
 {
-        Assert_MM_true(Discontiguous == getArrayLayout(objPtr));
+	Assert_MM_true(InlineContiguous != getArrayLayout(objPtr));
```

Comment:
Why this change? I think the old condition was more appropriate.

Reply:
Yes the old condition was more clear, however ever it does not account for `getArrayLayout` returning `Hybrid`, which should not trigger this assertion. Alternatively, perhaps this would be better:

`ArrayLayout layout = getArrayLayout(objPtr);`
`Assert_MM_true(Discontiguous == layout || (Hybrid == layout));`

Reply:
Yes, the old condition was more appropriate, however, it was incorrect as it did not account for array layout `Hydrid` which should not trigger the assert. However, perhaps this is best:

`ArrayLayout layout = getArrayLayout(objPtr);`
`Assert_MM_true((Discontiguous == layout) || (Hybrid == layout));`